posts/2021/04/rethinking-jupyter-documentation.md

```diff
@@ -0,0 +1,190 @@
+<!-- 
+.. title: Rethinking Jupyter Interactive Documentation
+.. slug: rethinking-jupyter-documentation
+.. date: 2021-04-12 11:59 UTC
+.. tags: Python, Open-Source, documentation
+.. category: 
+.. link: 
+.. description: 
+.. type: markdown
+.. author: Matthias Bussonnier
+-->
+
+Jupyter Notebook first release was 8 years ago – under the IPython Notebook
+name at the time. Even if notebooks were not invented by Jupyter; they were
+definitely democratized by it. Being Web powered allowed development of many
+changes in the Datascience world. Objects now often expose rich representation; from
+Pandas dataframes with as html tables, to more recent [Scikit-learn model](https://github.com/scikit-learn/scikit-learn/pull/14180).
+
+Today I want to look into a topic that has not evolved much since, and I believe
+could use an upgrade. Accessing interactive Documentation when in a Jupyter
+session, and what it could become. At the end I'll link to my current prototype
+if you are adventurous.
+
+<!-- TEASER_END -->
+
+# The current limitation for users
+
+The current documentation of IPython and Jupyter come in a few forms, but mostly
+have the same limitation.
+The typical way to reach for help is to use the `?` operator. Depending on
+the frontend you are using it will bring a pager, or a panel that will display
+some information about the current object.
+
+It can show some information about the current object (signature, file,
+sub/super classes) and the raw DocString of the object.
+
+You can scroll around but that's about it whether in terminal or Notebooks.
+
+Compare it to the same documentation on the numpy website.
+
+
+<img alt="numpy.linspace on numpy.org" src="/images/2021/04/numpy-linspace-compare.png" >
+
+On the left is the documentation for numpy when visiting https://numpy.org; on
+the right what you get in Jupyter Lab (or similar in a terminal).
```

Comment:
```suggestion
On the left is the documentation for numpy when visiting https://numpy.org. Let's
call that "rendered documentation". On the right what you get in Jupyter Lab or
in the IPython or regular Python REPL, let's cal that "help documentation" since
it is typically reached via `identifer?` or `help(identifier)`
```